.github/workflows/release.yml

```diff
@@ -62,32 +67,19 @@ jobs:
           path: |
             ${{ matrix.chain }}-runtime.wasm
             ${{ matrix.chain }}-srtool-digest.json
+
   build-docker:
     name: Build Docker
     runs-on: ubuntu-latest
     needs: build-binary
-    steps:
-      - name: Login to Docker Hub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          ref: ${{ github.event.inputs.tag }}
-      - name: Download binary
-        uses: actions/download-artifact@v2
-        with:
-          name: oak-collator
-      - name: Build
-        run: |
-          tag=${{ github.event.inputs.tag }}
-          docker_tag="${tag:1}"
-          docker build -f ./docker/turing/Dockerfile -t oaknetwork/turing:$docker_tag .
-          docker tag oaknetwork/turing:$docker_tag oaknetwork/turing:latest
-          docker push oaknetwork/turing:$docker_tag
-          docker push oaknetwork/turing:latest
+    use: ./.github/workflows/build-docker.yml
+    with:
+      tag: ${{ github.event.inputs.tag }}
+    secrets:
+      username: ${{ secrets.DOCKERHUB_USERNAME }}
+      password: ${{ secrets.DOCKERHUB_TOKEN }}
+    if: github.event.inputs.draft == 'true'
```

Comment:
Hmm I think we can remove it from this Release script, because the Docker publish job doesn’t need to run until the very end of the release, and as you said, the image is optional when upgrading Turing Network. The two jobs are not supposed to run simultaneously and have different priorities.